채널에서 채팅 글 좋아요 클릭 시, 중복해서 생성되는 이슈 해결

devjaewon opened the pull request:
# 채널에서 채팅 글 좋아요 클릭 시, 중복해서 생성되는 이슈 해결

## 해당 이슈 📎

- 채널에서 채팅 글 좋아요 클릭 시, 채팅글이 여러개 중복 생성되는 이슈 발생 (#99)

## 변경 사항 🛠

- `useQuery`를 활용하던 부분을 `useLazyQuery`를 이용해서 캐시 초기화시점을 컴포넌트 마운트 시점과 동기화

## 테스트 ✨

> 없음

## 리뷰어 참고 사항 🙋‍♀️

> 없음

jjazan committed:
[ BUGFIX ] 좋아요버튼 클릭시 채팅이 여러거 생겨나던 문제 해결 (#99)

Why
- 채팅 캐시를 불러와서 다시 쓰는 부분에 대해 불필요한 트리거가 생기고 있다

What
- useQuery 대신에 useLazyQuery 를 활용해서 캐시에 쓰는 부분을 정확히 컴포넌트가 마운트가 되는 시점으로 명시한다

happydhKim commented on web/src/hooks/channel/chat/useInitChat.js:
```diff
@@ -21,45 +21,46 @@ const GET_CHAT_LOGS = gql`
   };
   const writeChatCache = () => {
-    if (query.loading) return;
+    if (!query.called || query.loading) return;
```
```javascript
!query.called || query.loading
```
조건이 들어간 if문 같은 경우는 변수로 뺴는게 어떨까요?🤭

happydhKim commented on web/src/components/channel/Chat/Chat.jsx:
```diff
@@ -1,16 +1,19 @@
-import { useGetUserStatus, useChatChanged } from '@/hooks';
+import {
+  useChatChanged,
+  useInitChat,
```
3개 부터 내리는게 컨벤션 아니었나용?🤔

devjaewon replied:
아하 수정해야겠네요👍

happydhKim reviewed (approved):
빠르고 정확한 작업능력 언제나 감탄스러워요!!🥺

jjazan committed:
[ BUGFIX ] ?? 연타시 UI깨지던 이슈 해결 (#99)

Why
- ??? 나 !!! 와 같은 특수문자를 공백문자 없이 나열시 UI가 깨지던 이슈발생

What
- word-break: break-word 로 해결

jjazan committed:
[ REFACTOR ] 리뷰사항 반영 (#99)

Why
- 컨벤션에 맞지 않는 부분이 있음
- 가독성 있는 변수를 사용하는 것이 좋다

What
- import 문 한줄로 작성
- 필요한 변수명 작성하기
